hail/src/main/scala/is/hail/expr/ir/Parser.scala

```diff
@@ -330,6 +330,13 @@ object IRParser {
     struct_field(type_expr(env))(it)
   }
 
+  def ptype_exprs(env: TypeParserEnvironment)(it: TokenIterator): Array[PType] = {
+    punctuation(it, "(")
+    val types = repUntil(it, type_expr(env), PunctuationToken(")")).map(_.physicalType)
```

Comment:
I'm a little bit hung up on this line, as I'm not sure quite where this is going---is the intention to expose PTypes in python?

If not, how are we going to handle the IR serialization between python and Scala? Always convert to a default PType and then change if necessary?

Reply:
no ptype serialization between scala and python -- but I guess we've been trying to preserve the ability to do round trips with Pretty, right?

Reply:
I'll write a ptype serializer/deserializer separately, I think.

Reply:
So if I understand this correctly---the Python aggregations are going to create AggSignatures that use virtual types, and then Scala will convert them to ptypes and change as necessary?

Reply:
this is roughly right, but scala doesn't convert them to ptypes when it parses -- it converts them to ptypes in Extract (which should be a lowering pass soon), when we know that InferPtypes has already been run. 

The SeqOp node, for instance is not in Python, only scala. The physical signatures are only on the lower-level agg nodes.

Reply:
ah, that makes more sense. Thanks!